cmd/component-codegen/README.md

```diff
@@ -0,0 +1,56 @@
+# Component Codegen CLI
+
+## Overview
+`component-codegen` is a CLI tool built using the Cobra library to scaffold boilerplate code for new components. It simplifies the onboarding process by generating necessary files and updating relevant configurations.
+
+## Installation
+
+To build the CLI tool, run the following commands:
+```sh
+go mod tidy
+go build -o component-codegen <path to cli project>
+```
+
+Alternatively, use the Makefile available in the root project to streamline the build process:
+
+```sh
+make component-codegen
+```
+This will generate the binary in the `<project-root>/bin` folder.
+
+## Usage
+Once built, the CLI tool can be used to create new components like below:
```

Comment:
I would leave UI on that level https://github.com/opendatahub-io/opendatahub-operator/pull/1488#pullrequestreview-2550796138

Reply:
The UI already has that, this was just additional info
```
./bin/component-codegen generate --help                                                                         ─╯
Generates boilerplate folders/files for a new component

Usage:
  codegen-cli generate [component-name] [flags]

Flags:
  -h, --help   help for generate
  ```

Reply:
I mean, for user to announce only make wrapper to not bother with compilation of the actual generator. It leaves freedom to change layout and also makes makefile more clear, e.g:

```
new-component: $(LOCALBIN)/component-codegen
    $< generate $(COMPONENT)

$(LOCALBIN)/component-codegen:
    go build -o $@
```

Reply:
thanks for the explanation @ykaliuta  and help :) Updated the code now